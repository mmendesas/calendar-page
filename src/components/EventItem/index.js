import React from 'react';
import PropTypes from 'prop-types';
import IconLocation from '../../assets/images/icon-location.png';

import {
  Container,
  Content,
  Flag,
  Title,
  DeadlineInfo,
  Location,
  Button,
} from './styles';

export default function EventItem({ data }) {
  const { image, title, datetime, location, deadline } = data;

  return (
    <Container>
      <img src={image} alt="event" />
      <Flag name={location.country.toLowerCase()} />
      <Content>
        <Title>{title}</Title>
        <div>{datetime}</div>
        <Location>
          <img src={IconLocation} alt="icon-location" />
          <div>
            {location.city} - {location.country}
          </div>
        </Location>
        <DeadlineInfo>
          <strong>Deadline:</strong>
          <span>{deadline}</span>
        </DeadlineInfo>
        <Button>See Details</Button>
      </Content>
    </Container>
  );
}

EventItem.propTypes = {
  data: PropTypes.shape({
    image: PropTypes.string,
    title: PropTypes.string,
    datetime: PropTypes.string,
    location: PropTypes.shape({
      city: PropTypes.string,
      country: PropTypes.string,
    }),
    deadline: PropTypes.string,
  }).isRequired,
};
