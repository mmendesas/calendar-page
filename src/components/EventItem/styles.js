import styled from 'styled-components';

export const Container = styled.li`
  background-color: rgb(255, 255, 255);
  width: 300px;
  min-height: 324px;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 0px 6px 0px,
    rgba(0, 0, 0, 0.1) 0px 2px 12px 0px;
  border-radius: 4px;
  overflow: hidden;
  position: relative;

  > img {
    background-size: cover;
    height: 150px;
  }
`;

export const Content = styled.div`
  display: flex;
  flex-direction: column;
  margin: 16px;
  line-height: 1.5;
`;

export const Title = styled.div`
  font-weight: 500;
  font-size: 20px;
  padding-bottom: 5px;
  line-height: 1.6;
`;

export const DeadlineInfo = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  > span {
    padding-left: 5px;
  }
`;

export const Location = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  font-size: 14px;
`;

export const Button = styled.button`
  background-color: #3c74cf;
  color: white;
  padding: 8px;
  box-shadow: 0px 1px 5px 0px rgba(0, 0, 0, 0.2),
    0px 2px 2px 0px rgba(0, 0, 0, 0.14), 0px 3px 1px -2px rgba(0, 0, 0, 0.12);
  font-weight: bold;
  border-radius: 4px;
  letter-spacing: 0.02857em;
  /* text-transform: uppercase; */
  width: 110px;
  align-self: flex-end;

  &:hover {
    background-color: rgb(42, 81, 144);
  }
`;

export const Flag = styled.div`
  background: url(https://vanhackblobstorageprod.blob.core.windows.net/flags/${props => props.name || 'brazil'}.svg)
    no-repeat center;
  width: 30px;
  height: 25px;
  border-radius: 50%;
  position: absolute;
  top: 135px;
  right: 10px;
  z-index: 1;
`;
