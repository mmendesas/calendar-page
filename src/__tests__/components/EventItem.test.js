import React from 'react';
import { render } from '@testing-library/react';

import EventItem from '~/components/EventItem';

describe('[Component] EventItem', () => {
  const setup = () => {
    const info = {
      title: 'Some Test',
      datetime: 'November 15 - 20, 2019',
      location: {
        city: 'SaoPaulo',
        country: 'Brazil',
      },
      deadline: '11/20/2019',
      image:
        'https://vanhackblobstorageprod.blob.core.windows.net/img/events/thumbnail/37478757-eef9-4cf1-b0c2-9ceb426cffaa.jpg',
    };
    const utils = render(<EventItem data={info} />);

    return {
      info,
      utils,
    };
  };

  it('should be match snapshot', () => {
    const { utils } = setup();
    const { getByTestId } = utils;
    expect(getByTestId('eventItem')).toMatchSnapshot();
  });
});
