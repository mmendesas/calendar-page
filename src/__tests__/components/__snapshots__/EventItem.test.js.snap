// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`[Component] EventItem should be match snapshot 1`] = `
<li
  class="sc-bxivhb floOoc"
  data-testid="eventItem"
>
  <img
    alt="event"
    src="https://vanhackblobstorageprod.blob.core.windows.net/img/events/thumbnail/37478757-eef9-4cf1-b0c2-9ceb426cffaa.jpg"
  />
  <div
    class="sc-dnqmqq hWwYNg"
    name="brazil"
  />
  <div
    class="sc-ifAKCX APnor"
  >
    <div
      class="sc-EHOje SoJTC"
    >
      Some Test
    </div>
    <div>
      November 15 - 20, 2019
    </div>
    <div
      class="sc-gzVnrw fKazfQ"
    >
      <img
        alt="icon-location"
        src="icon-location.png"
      />
      <div>
        SaoPaulo
         - 
        Brazil
      </div>
    </div>
    <div
      class="sc-bZQynM eYzvQb"
    >
      <strong>
        Deadline:
      </strong>
      <span>
        11/20/2019
      </span>
    </div>
    <button
      class="sc-htoDjs fWqFVQ"
    >
      See Details
    </button>
  </div>
</li>
`;
